src/python/pants/testutil/BUILD

```diff
@@ -24,7 +24,6 @@ python_sources(
         "pants_integration_test.py": {
             "dependencies": ["//BUILD_ROOT:files", "src/python/pants/__main__.py"]
```

Comment:
I also tried removing the `//BUILD_ROOT:files` dependency from `pants_integration_test.py`. But that didn't work, and it turned out to be a more invasive change.

Fixing this will require removing changing tmpdir creation functions in this file that call `get_buildroot()` repeatedly. We need to change that assumption so the test author can pass in the build root instead of having these functions calculate it from the current working directory, which happens to be the pytest sandbox.

I suspect this issue existed before #20887 (ie it is not a new bug in 2.22.x), but `pants_integration_test`-based tests are far less common than `RuleRunner` based tests. Since this is (probably) not a new bug, we can deal with changing the assumptions in this code at some later date.